server/src/http/broadcaster.rs

```diff
@@ -101,88 +124,88 @@ impl Broadcaster {
     async fn heartbeat(&self) {
         let mut active_connections = 0i64;
         for mut group in self.active_connections.iter_mut() {
-            let mut ok_clients = Vec::new();
+            let clients = std::mem::take(&mut group.clients);
+            let ok_clients = &mut group.clients;
 
-            for client in &group.clients {
-                if client
+            for ClientData { token, sender } in clients {
+                if sender
                     .send(sse::Event::Comment("keep-alive".into()))
                     .await
                     .is_ok()
                 {
-                    ok_clients.push(client.clone());
+                    ok_clients.push(ClientData { token, sender });
                 }
             }
 
             active_connections += ok_clients.len() as i64;
-            group.clients = ok_clients;
         }
         CONNECTED_STREAMING_CLIENTS.set(active_connections)
     }
 
-    /// Registers client with broadcaster, returning an SSE response body.
     pub async fn connect(
         &self,
         token: EdgeToken,
-        filter_set: Query<FeatureFilters>,
-        query: unleash_types::client_features::Query,
+        query: Query,
     ) -> EdgeResult<Sse<InfallibleStream<ReceiverStream<sse::Event>>>> {
-        let (tx, rx) = mpsc::channel(10);
+        self.create_connection(StreamingQuery::from((&query, &token)), &token.token)
+            .await
+            .map(Sse::from_infallible_receiver)
+    }
 
-        let features = &self
-            .resolve_features(&token, filter_set.clone(), query.clone())
-            .await?;
+    async fn create_connection(
```

Comment:
I split `connect` into `connect` and `create_connection` to facilitate easier testing. Instead of passing in an EdgeToken and a Query (which are almost the same thing), I wanted to just pass the bare minimum that we needed.